src/Microsoft.VisualStudio.Web.CodeGenerators.Mvc/Controller/ControllerGeneratorBase.cs

```diff
@@ -83,13 +84,31 @@ protected virtual IEnumerable<string> TemplateFolders
             }
         }
 
-        protected string GetControllerNamespace()
+        protected string GetDefaultControllerNamespace()
         {
             // Review: MVC scaffolding used ActiveProject's MSBuild RootNamespace property
             // That's not possible in command line scaffolding - the closest we can get is
             // the name of assembly??
             var appName = ApplicationInfo.ApplicationName;
-            return appName + "." + Constants.ControllersFolderName;
+
+            var defaultControllerNamespace = appName + "." + Constants.ControllersFolderName;
+            
```

Comment:
I don't think looking for a .xproj file is the right approach. Agreed that by convention, it would be in the same folder as project.json and we could do the above heuristic approach to find the xproj when there are multiple. But unless we have a foolproof way of telling that a xproj is the correct one and the current intended one, using it is as good as guessing that it *might* be what user intended. 

The whole idea behind exposing the controllerNamespace is two fold: 
1. Scaffolding currently is still intended to be a Visual Studio feature and Visual studio already has mechanisms of finding the right namespace.
2. It also gives command line users the flexibility to specify the namespace if they do not want to follow the default pattern. 

---
In reply to: [64578045](https://github.com/aspnet/Scaffolding/pull/238#discussion_r64578045) [](ancestors = 64578045)